From 96f315bf4e52b445d0bedfeee1df66a96a96223c Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Fri, 27 Oct 2017 19:33:43 -0700
Subject: [PATCH 11/11] libbt: Only allow upio_start_stop_timer on 32bit arm

This apparently isn't exposed on aarch64 kernels and is thus
causing a null-pointer dereference

Change-Id: I0d4373a4c1002ba4a0551e2fcc120a1ab12c9138
---
 src/upio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/upio.c b/src/upio.c
index d24d496..7d268b8 100644
--- a/src/upio.c
+++ b/src/upio.c
@@ -499,7 +499,7 @@ void upio_set(uint8_t pio, uint8_t action, uint8_t polarity)
                 ALOGE("upio_set : write(%s) failed: %s (%d)",
                         VENDOR_BTWRITE_PROC_NODE, strerror(errno),errno);
             }
-#if (PROC_BTWRITE_TIMER_TIMEOUT_MS != 0)
+#if !defined(__aarch64__) && (PROC_BTWRITE_TIMER_TIMEOUT_MS != 0)
             else
             {
                 /* arm user space timer based on action */
-- 
2.7.4

